engine/conditions.go

```diff
@@ -39,23 +44,34 @@ type ExternalLinks struct {
 }
 
 type Conditional interface {
-	Evaluate(log logr.Logger, ctx map[string]interface{}) (ConditionResponse, error)
+	Evaluate(log logr.Logger, ctx ConditionContext) (ConditionResponse, error)
 }
 
 type Rule struct {
 	RuleID      string      `yaml:"ruleID,omitempty"`
 	Description string      `yaml:"description,omitempty"`
 	Category    string      `yaml:"category,omitempty"`
-	Perform     string      `yaml:"perform,omitempty"`
+	Perform     Perform     `yaml:"perform,omitempty"`
```

Comment:
I'd make this an `inline`  so we don't need to add the perform keyword in the yaml